File: plugins/agento11y/internal/agents/cursor/tags/tags_test.go
package tags

import "testing"

func TestBuild(t *testing.T) {
	cases := []struct {
		name  string
		in    BuiltinInputs
		check func(t *testing.T, got map[string]string)
	}{
		{
			name: "git branch and explicit cwd populated",
			in: BuiltinInputs{
				WorkspaceRoot: "/ws",
				Cwd:           "/real-cwd",
				GitBranch:     "main",
			},
			check: func(t *testing.T, got map[string]string) {
				if got["git.branch"] != "main" {
					t.Errorf("git.branch = %q; want main", got["git.branch"])
				}
				if got["cwd"] != "/real-cwd" {
					t.Errorf("cwd = %q; want /real-cwd", got["cwd"])
				}
			},
		},
		{
			name: "cwd falls back to workspace root",
			in:   BuiltinInputs{WorkspaceRoot: "/ws"},
			check: func(t *testing.T, got map[string]string) {
				if got["cwd"] != "/ws" {
					t.Errorf("cwd should fall back to workspace root; got %q", got["cwd"])
				}
			},
		},
		{
			name: "no inputs returns nil",
			in:   BuiltinInputs{},
			check: func(t *testing.T, got map[string]string) {
				if got != nil {
					t.Errorf("Build with no inputs must return nil; got %v", got)
				}
			},
		},
		{
			name: "subagent set when background",
			in:   BuiltinInputs{IsBackgroundAgent: true, WorkspaceRoot: "/ws"},
			check: func(t *testing.T, got map[string]string) {
				if got["subagent"] != "true" {
					t.Errorf("subagent should be set; got %q", got["subagent"])
				}
			},
		},
		{
			name: "subagent absent when not background",
			in:   BuiltinInputs{WorkspaceRoot: "/ws"},
			check: func(t *testing.T, got map[string]string) {
				if _, ok := got["subagent"]; ok {
					t.Errorf("subagent should be absent for non-background agent")
				}
			},
		},
	}
	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
			tc.check(t, Build(tc.in))
		})
	}
}

File: plugins/opencode/src/tags.test.ts
import { describe, expect, it } from "vitest";
import { buildBuiltinTags } from "./tags.js";

describe("buildBuiltinTags", () => {
  const cases: {
    name: string;
    in: Parameters<typeof buildBuiltinTags>[0];
    want: ReturnType<typeof buildBuiltinTags>;
  }[] = [
    {
      name: "both keys populated",
      in: { cwd: "/repo", gitBranch: "main" },
      want: { "git.branch": "main", cwd: "/repo" },
    },
    {
      name: "branch only",
      in: { gitBranch: "main" },
      want: { "git.branch": "main" },
    },
    {
      name: "cwd only",
      in: { cwd: "/repo" },
      want: { cwd: "/repo" },
    },
    {
      name: "empty inputs return undefined",
      in: {},
      want: undefined,
    },
    {
      name: "empty-string inputs return undefined",
      in: { cwd: "", gitBranch: "" },
      want: undefined,
    },
  ];

  it.each(cases)("$name", ({ in: input, want }) => {
    expect(buildBuiltinTags(input)).toEqual(want);
  });
});
